source/common/network/utility.cc

```diff
@@ -208,6 +208,25 @@ Address::InstanceConstSharedPtr Utility::getLocalAddress(const Address::IpVersio
   return ret;
 }
 
+bool Utility::isLocalConnection(const Address::Instance& local_address,
+                                const Address::Instance& remote_address) {
+  auto local_ip = local_address.ip();
+  auto remote_ip = remote_address.ip();
+  if (local_ip && remote_ip) {
+    if (local_ip->ipv4() && remote_ip->ipv4()) {
+      return local_ip->ipv4()->address() == remote_ip->ipv4()->address();
```

Comment:
What will you do if the remote host has 127.* address from the loopback address space ?

Reply:
@rshriram 

> This is a brittle model. Local and non local traffic can be distinguished using a simple not condition in the source match and matching against 127.*..

This is a much more resilient model. Local and remote traffic cannot be distinguished by checking for 127... Because if the machine has a network interface with 10.0.0.1   IP then connections to 10.0.0.1 from the same host will not match the 127.... condition.

We have had this implemented in-house for years and found out that the most resilient solution is to check for matching source and destination address. This handles multiple network interfaces, as well as multiple IP addresses on the interfaces and having different protocols.

This feature will work perfectly fine with the source ip match when implemented. It will be able to narrow down on the SAME_HOST and EXTERNAL_HOST. It will  will allow to distinguish for example (SAME_HOST on 10.0.0.1) from (EXTERNAL_HOST on 10.0.0.1) from (SAME_HOST on 127.**)

Reply:
> What will you do if the remote host has 127.* address from the loopback address space ?
It will be handled correctly. The network stack will guarantee that the source and the destination ip address will be from the 127.* and that they will match.

Reply:
As a result of this discussion I have renamed the enum constants as advised by @rshriram and extended the relevant doc strings. There's a unit test to verify `isLocalConnection`.